### PR TITLE
BUG: Index sortlevel ascending add type checking #32334

### LINKED ISSUE #32334: DataFrame.sort_index() with ascending passed as a list on a single level index gives wrong result
I found some problems while using data_frame sort_index,  first, create dataframe

```
data = pd.DataFrame(np.array([[1, 2, 3], [4, 5, 6], [7, 8, 9]]), columns=['a', 'b', 'c'])
data = data.set_index('a')
data.sort_index(level=['a'], ascending=[False])
```
Then I want to sort index "a" in descending order,
```
data.sort_index(level=['a'], ascending=[False])
```
The above code does not sort index a in descending order,   I found the reason is that this dataframe is a single index structure, so when ascending = list is passed, it does not take effect

https://github.com/pandas-dev/pandas/blob/bf613c14f41b624b432d0d6b9a29007e7990d460/pandas/core/frame.py#L4777-L4783

Because when it is used, the index may be single index or multi-index. In order to use the effect uniformly,I think there should be a judgment here. If index is a single index and ascending type == list, ascending should be equal to ascending [0]，This may be more user friendly

If yes, can I mention pr

### PR DESCRIPTION
**Pull request by aniaan**
- [x] closes #32334
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

Add the check of ascending parameter in the sortlevel method, only allow bool value and single bool value list

cc @Dr-Irv

### CODE REVIEW

**Review by Dr-Irv — CHANGES_REQUESTED** [review]
Looks good.  Can you add a `whatsnew` entry for v1.2 ?

**Comment by BeanNan** [comment]
> Looks good. Can you add a `whatsnew` entry for v1.2 ?

I have added the v1.2 whatsnew entry , please take a look

File: doc/source/whatsnew/v1.2.0.rst [hunk 1]
```diff
@@ -352,7 +352,7 @@ Indexing
 - Bug in :meth:`PeriodIndex.get_loc` incorrectly raising ``ValueError`` on non-datelike strings instead of ``KeyError``, causing similar errors in :meth:`Series.__geitem__`, :meth:`Series.__contains__`, and :meth:`Series.loc.__getitem__` (:issue:`34240`)
 - Bug in :meth:`Index.sort_values` where, when empty values were passed, the method would break by trying to compare missing values instead of pushing them to the end of the sort order. (:issue:`35584`)
 - Bug in :meth:`Index.get_indexer` and :meth:`Index.get_indexer_non_unique` where int64 arrays are returned instead of intp. (:issue:`36359`)
--
+- Bug in :meth:`Index.sortlevel` where, add the check of ascending parameter in the sortlevel method, only allow bool value and single bool value list. (:issue:`32334`)
```
Comment (Dr-Irv) [inline doc/source/whatsnew/v1.2.0.rst:355]: Better to say

```
Bug in :meth:`DataFrame.sort_index` where parameter ascending passed as a list on a single level index gives wrong result. (:issue:`32334`)
```

**Review by Dr-Irv — CHANGES_REQUESTED** [review]

File: doc/source/whatsnew/v1.2.0.rst (hunk 1 above)
Comment (BeanNan) [reply doc/source/whatsnew/v1.2.0.rst]: whatsnew has been adjusted, please take a look

**Review by Dr-Irv — APPROVED** [review]

**Comment by Dr-Irv** [comment]
thanks @BeanNan

File: pandas/tests/indexes/test_base.py [hunk 2]
```diff
@@ -2222,6 +2222,31 @@ def test_contains_method_removed(self, index):
             with pytest.raises(AttributeError, match=msg):
                 index.contains(1)
 
+    def test_sortlevel(self):
+        index = pd.Index([5, 4, 3, 2, 1])
```
Comment (jreback) [inline pandas/tests/indexes/test_base.py:2226]: we generally add the issue number here as a comment

File: pandas/tests/indexes/test_base.py [hunk 3]
```diff
@@ -2222,6 +2222,31 @@ def test_contains_method_removed(self, index):
...
+
+        with pytest.raises(
+            Exception, match="ascending must be a list of bool values of length 1"
+        ):
+            index.sortlevel(ascending=[True, True])
+
+        with pytest.raises(Exception, match="ascending must be a bool value"):
+            index.sortlevel(ascending=["True"])
+
+        expected = pd.Index([1, 2, 3, 4, 5])
```
Comment (jreback) [inline pandas/tests/indexes/test_base.py:2238]: this should be 2 tests, one for the exceptions, one for the working cases (which could be parameterized)

File: pandas/core/indexes/base.py [hunk 4]
```diff
@@ -1515,6 +1515,20 @@ def sortlevel(self, level=None, ascending=True, sort_remaining=None):
         -------
         Index
         """
+        if not isinstance(ascending, (list, bool)):
```
Comment (jreback) [inline pandas/core/indexes/base.py:1518]: these should be checked at a much lower level, where they are used; the reason these are fairly generic things not specific to Index in this case

**Review by jreback — COMMENTED** [review]
@Dr-Irv ideally this would have had more review

**Pull request merged**